sdk/python/kfp_tekton/compiler/_op_to_template.py

```diff
@@ -106,21 +105,81 @@ def _op_to_template(op: BaseOp):
         }
 
     elif isinstance(op, dsl.ResourceOp):
-        # # no output artifacts
-        # output_artifacts = []
-        #
-        # # workflow template
-        # processed_op.resource["manifest"] = yaml.dump(
-        #     convert_k8s_obj_to_json(processed_op.k8s_resource),
-        #     default_flow_style=False
-        # )
-        # template = {
-        #     'name': processed_op.name,
-        #     'resource': convert_k8s_obj_to_json(
-        #         processed_op.resource
-        #     )
-        # }
-        raise NotImplementedError("dsl.ResourceOp is not yet implemented")
+        # no output artifacts
+        output_artifacts = []
+
+        # task template
+        template = {
+            'apiVersion': tekton_api_version,
+            'kind': 'Task',
+            'metadata': {'name': processed_op.name},
+            'spec': {
+                "params": [
+                    {
+                        "description": "Action on the resource",
+                        "name": "action",
+                        "type": "string"
+                    },
+                    {
+                        "default": "strategic",
+                        "description": "Merge strategy when using action patch",
+                        "name": "merge-strategy",
+                        "type": "string"
+                    },
+                    {
+                        "description": "Content of the resource to deploy",
+                        "name": "manifest",
+                        "type": "string"
+                    },
+                    {
+                        "default": "",
+                        "description": "An express to retrieval data from resource.",
+                        "name": "output",
+                        "type": "string"
+                    },
+                    {
+                        "default": "",
+                        "description": "A label selector express to decide if the action on resource is success.",
+                        "name": "success-condition",
+                        "type": "string"
+                    },
+                    {
+                        "default": "",
+                        "description": "A label selector express to decide if the action on resource is failure.",
+                        "name": "failure-condition",
+                        "type": "string"
+                    },
+                    {
+                        "default": "index.docker.io/fenglixa/kubeclient:v0.0.1",
```

Comment:
Can also add a todo comment to replace this image once our task is pushed to Tekton catalog?

Reply:
Addressed. Thanks @Tomcli